crates/runtime/src/accelerated_table/refresh_task/changes.rs

```diff
@@ -100,16 +105,22 @@ impl RefreshTask {
                             }
                         }
                         Err(e) => {
-                            self.mark_dataset_status(status::ComponentStatus::Error)
-                                .await;
+                            Self::mark_dataset_status(
```

Comment:
We are not using sql value from above as it can be changed dynamically? https://github.com/spiceai/spiceai/pull/2515/files#diff-9fb43c580ce020358c24d68b4e431efbee2d5583820210dd3c06d499b985cb74R70

Reply:
That was my initial thought yes. I could see us moving it to L82 (at the start of `Ok(change_envelope) => {`), so that we write the correct SQL to error metrics if an error occurs.